_pytest/python.py

```diff
@@ -1246,6 +1260,11 @@ def __exit__(self, *tp):
         suppress_exception = issubclass(self.excinfo.type, self.expected_exception)
         if sys.version_info[0] == 2 and suppress_exception:
             sys.exc_clear()
+        if self.match_expr:
+            try:
```

Comment:
Hmm this bare try/except makes me a little uneasy... why is it needed? Wouldn't be enough to just call `self.excinfo.match(self.match_expr)`?

```python
if self.match_expr:
    self.excinfo.match(self.match_expr)
```